server/index.js

```diff
@@ -50,7 +50,7 @@ export default class Server {
 
   handleRequest (req, res, parsedUrl) {
     // Parse url if parsedUrl not provided
-    if (!parsedUrl) {
+    if (!parsedUrl || typeof parsedUrl !== 'object') {
```

Comment:
This is a good find. I like it.
But instead of creating the parsing the url, we should throw an error with a good message.

Because this is a mistake. With the given example, parsedUrl will be a function which is the function to call the next middleware in express.

So, that's a mistake.